awx_collection/plugins/modules/tower_organization.py

```diff
@@ -66,49 +77,56 @@
     tower_config_file: "~/tower_cli.cfg"
 '''
 
-from ..module_utils.ansible_tower import TowerModule, tower_auth_config, tower_check_mode
-
-try:
-    import tower_cli
-    import tower_cli.exceptions as exc
-
-    from tower_cli.conf import settings
-except ImportError:
-    pass
+from ..module_utils.tower_api import TowerModule
 
 
 def main():
+    # Any additional arguments that are not fields of the item can be added here
     argument_spec = dict(
-        name=dict(required=True),
-        description=dict(),
+        name=dict(type='str', required=True),
+        description=dict(type='str', required=False),
         custom_virtualenv=dict(type='str', required=False),
-        state=dict(choices=['present', 'absent'], default='present'),
+        max_hosts=dict(type='int', required=False, default="0"),
+        state=dict(type='str', choices=['present', 'absent'], default='present', required=False),
     )
 
+    # Create a module for ourselves
     module = TowerModule(argument_spec=argument_spec, supports_check_mode=True)
 
+    # Extract our parameters
     name = module.params.get('name')
     description = module.params.get('description')
     custom_virtualenv = module.params.get('custom_virtualenv')
+    max_hosts = module.params.get('max_hosts')
+    # instance_group_names = module.params.get('instance_groups')
     state = module.params.get('state')
 
-    json_output = {'organization': name, 'state': state}
-
-    tower_auth = tower_auth_config(module)
-    with settings.runtime_values(**tower_auth):
-        tower_check_mode(module)
-        organization = tower_cli.get_resource('organization')
-        try:
-            if state == 'present':
-                result = organization.modify(name=name, description=description, custom_virtualenv=custom_virtualenv, create_on_missing=True)
-                json_output['id'] = result['id']
-            elif state == 'absent':
-                result = organization.delete(name=name)
-        except (exc.ConnectionError, exc.BadRequest, exc.AuthError) as excinfo:
-            module.fail_json(msg='Failed to update the organization: {0}'.format(excinfo), changed=False)
-
-    json_output['changed'] = result['changed']
-    module.exit_json(**json_output)
+    # Attempt to look up the related items the user specified (these will fail the module if not found)
+    # instance_group_objects = []
+    # for instance_name in instance_group_names:
+    #     instance_group_objects.append(module.resolve_name_to_id('instance_groups', instance_name))
```

Comment:
We could file a new issue for org instance groups if there isn't one already. Let's get rid of the comments for getting this wrapped up.